sirepo/job_api.py

```diff
@@ -95,6 +95,36 @@ def api_downloadDataFile(simulation_type, simulation_id, model, frame, suffix=No
         )
 
 
+@api_perm.require_user
+def api_exportRSOptConfig(simulation_type, simulation_id, filename):
+    import zipfile
```

Comment:
We are trying to move away from running code inside the server to inside the agent. I think this is a good example of code that should be run within the agent. From the looks of the code I think it should be a call to [statefulCompute](https://github.com/radiasoft/sirepo/wiki/Interacting-with-Simulations-Architecture#b-stateful-compute). You can see an example of this type of call [here](https://github.com/radiasoft/sirepo/blob/f45197e925b068846a96f6fa4b15c35640ad07e3/sirepo/package_data/static/js/controls.js#L197). 

One thing that will need to change is that the call to [`lib_files_for_export`](https://github.com/radiasoft/sirepo/blob/ec7ef734012eda4937139efc63fa367df48252c7/sirepo/template/srw.py#L400) will fail because that call is protected by [`sim_data._assert_server_side()`](https://github.com/radiasoft/sirepo/blob/ec7ef734012eda4937139efc63fa367df48252c7/sirepo/sim_data/__init__.py#L368) and we will be in the agent not in the server. @robnagler can you remind me why we have calls to assert_server_side()? Was it before we had ability to pull over lib files?

Reply:
I thought job apis were run in the agent.  What distinguishes "agent" from "server"?

Reply:
`job_api` just contains api endpoints in the flask server that _should_ be passed on to the supervisor (who will then pass onto the agent if the api is destined for the agent). The api function defined in job_api needs to call the [proper code](https://github.com/radiasoft/sirepo/blob/f45197e925b068846a96f6fa4b15c35640ad07e3/sirepo/job_api.py#L245) to get the request forward onto the supervisor.

That help? 

is the entrypoint from calls to the server to be passed to the supervisor who will then pass to the agent.

Reply:
So a request destined for an agent goes on a journey like: GUI -> flask server (sirepo.job_api) -> supervisor (pkcli.job_supervisor -> sirepo.job_supervisor) -> job_driver (whichever type is configured / needed) -> job_agent